addons/product/product.py

```diff
@@ -926,29 +926,33 @@ def _get_price_extra(self, cr, uid, ids, name, args, context=None):
         return result
 
     def _select_seller(self, cr, uid, product_id, partner_id=False, quantity=0.0, date=time.strftime(DEFAULT_SERVER_DATE_FORMAT), uom_id=False, context=None):
-        if context is None:
-            context = {}
-        res = self.pool.get('product.supplierinfo').browse(cr, uid, [])
-        for seller in product_id.seller_ids:
-            # Set quantity in UoM of seller
-            quantity_uom_seller = quantity
-            if quantity_uom_seller and uom_id and uom_id != seller.product_uom:
-                quantity_uom_seller = uom_id._compute_qty_obj(uom_id, quantity_uom_seller, seller.product_uom)
-
-            if seller.date_start and seller.date_start > date:
-                continue
-            if seller.date_end and seller.date_end < date:
-                continue
-            if partner_id and seller.name not in [partner_id, partner_id.parent_id]:
-                continue
-            if quantity_uom_seller < seller.qty:
-                continue
-            if seller.product_id and seller.product_id != product_id:
-                continue
 
-            res |= seller
-            break
-        return res
+        def _find_seller(sellers):
```

Comment:
The reason for changes is that, When we create an RFQ : the system fetches the variant cost price (from vendor O2m), if no price on variant, it takes the vendor price from template.